Assets/Scripts/CPURodInputHandler.cs

```diff
@@ -0,0 +1,96 @@
+using UnityEngine;
+
+public class CPURodInputHandler : IRodInputHandler
+{
+    private Transform _ballTransform;
+    private Transform _rodTransform;
+    private Doll[] _dolls;
+
+    private float _moveSpeed;
+    private float _rotationSpeed;
+    private const float REACTION_DISTANCE = 1.5f; // ボールに反応する最大距離
+    private const float ROTATION_MULTIPLIER = 300f;
+
+    public CPURodInputHandler(Transform ballTransform, Transform rodTransform, Doll[] dolls, float moveSpeed, float rotationSpeed)
+    {
+        _ballTransform = ballTransform;
+        _rodTransform = rodTransform;
+        _dolls = dolls;
+        _moveSpeed = moveSpeed;
+        _rotationSpeed = rotationSpeed * ROTATION_MULTIPLIER;
+    }

```

Comment:
引数の型がTransformだと、間違って別のオブジェクトが入る可能性がある。
BallとRodControllerで受けて、コンストラクタ内でball.transformを取得する方が良いかも。
あるいは、positionの情報しか使ってないので、BallクラスにGetPosition()を実装して、Ballクラスでインスタンスを保持するのもあり。
同様にRodControllerクラスにGetCurrentPosition()、GetCurrentRotation()を実装して、そっちから取得する。
DollsもRodControllerから取得できるようにすれば、引数1つ減らせる。

（任意）
ModeSettingsを読み取り専用にしたので、難易度設定の部分だけ別メソッドにまとめて、settingsごと渡しても良いかも。
点差に応じて試合中に難易度を自動変更したりできるようになる。負けそうになるとちょっと強くなるとか。
```suggestion
    public CPURodInputHandler(Ball ball, RodController rodController)
    {
        _ball = ball;
        _rodController = rodController;
        _dolls = rodController.GetDolls();
    }

    public void ApplyCPUSettings(CPUConfig.ModeSettings settings)
    {
        _moveSpeed = settings.MoveSpeed;
        _rotationSpeed = settings.RotationSpeed * ROTATION_MULTIPLIER;
    }
```

Reply:
RodControllerクラスのGetCurrentPosition()、GetCurrentRotation()がうまくいかなかったので、一旦`_rodController.transform.position`みたいに取得してます